tests/subsys/audio_module/src/bad_param_test.c

```diff
@@ -0,0 +1,1171 @@
+	zassert_equal(ret, -ECANCELED,
+		      "Configuration get function did not return -ECANCELED (%d): ret %d",
+		      -ECANCELED, ret);
+}
+
+ZTEST(suite_audio_module_bad_param, test_config_get_null)
+{
+	int ret;
+
+	test_initialise_description(&test_description_rx, "Module Test",
+				    AUDIO_MODULE_TYPE_UNDEFINED, NULL);
+
+	test_initialise_handle(&handle, "TEST Get Config", &test_description,
+			       AUDIO_MODULE_STATE_UNDEFINED, NULL, NULL);
+
+	ret = audio_module_configuration_get(NULL, NULL);
+	zassert_equal(ret, -EINVAL,
+		      "Configuration get function did not return -EINVAL (%d): ret %d", -EINVAL,
+		      ret);
+
+	ret = audio_module_configuration_get(NULL, config);
+	zassert_equal(ret, -EINVAL,
+		      "Configuration get function did not return -EINVAL (%d): ret %d", -EINVAL,
+		      ret);
+
+	ret = audio_module_configuration_get(&handle, NULL);
+	zassert_equal(ret, -EINVAL,
+		      "Configuration get function did not return -EINVAL (%d): ret %d", -EINVAL,
+		      ret);
+
+	handle.state = AUDIO_MODULE_STATE_STOPPED;
+	ret = audio_module_configuration_get(NULL, config);
+	zassert_equal(ret, -EINVAL,
+		      "Configuration get function did not return -EINVAL (%d): ret %d", -EINVAL,
+		      ret);
+	zassert_equal(handle.state, AUDIO_MODULE_STATE_STOPPED,
+		      "Configuration get returns with incorrect state: %d", handle.state);
+
+	handle.state = AUDIO_MODULE_STATE_STOPPED;
+	ret = audio_module_configuration_get(&handle, NULL);
+	zassert_equal(ret, -EINVAL,
+		      "Configuration get function did not return -EINVAL (%d): ret %d", -EINVAL,
+		      ret);
+	zassert_equal(handle.state, AUDIO_MODULE_STATE_STOPPED,
+		      "Configuration get returns with incorrect state: %d", handle.state);
+
+	handle.state = AUDIO_MODULE_STATE_STOPPED;
+	ret = audio_module_configuration_get(NULL, NULL);
+	zassert_equal(ret, -EINVAL,
+		      "Configuration get function did not return -EINVAL (%d): ret %d", -EINVAL,
+		      ret);
+	zassert_equal(handle.state, AUDIO_MODULE_STATE_STOPPED,
+		      "Configuration get returns with incorrect state: %d", handle.state);
+}
+
+ZTEST(suite_audio_module_bad_param, test_reconfig_bad_state)
+{
+	int ret;
+
+	test_initialise_description(&test_description_rx, "Module Test",
+				    AUDIO_MODULE_TYPE_UNDEFINED, NULL);
+
+	test_initialise_handle(&handle, "TEST Reconfig", &test_description,
+			       AUDIO_MODULE_STATE_UNDEFINED, NULL, NULL);
+
+	ret = audio_module_reconfigure(&handle, config);
+	zassert_equal(ret, -ECANCELED,
+		      "Reconfiguration function did not return -ECANCELED (%d): ret %d", -ECANCELED,
+		      ret);
+}
+
+ZTEST(suite_audio_module_bad_param, test_reconfig_null)
+{
+	int ret;
+
+	test_initialise_description(&test_description_rx, "Module Test",
+				    AUDIO_MODULE_TYPE_UNDEFINED, NULL);
+
+	test_initialise_handle(&handle, "TEST Reconfig", &test_description,
+			       AUDIO_MODULE_STATE_CONFIGURED, NULL, NULL);
+
+	ret = audio_module_reconfigure(NULL, NULL);
```

Comment:
In this test, it seems that "almost" the same thing is tested twice?

Reply:
It is almost the same, but is different as both pointers are NULL. I could remove this test and all similar from the other tests?